tests/tests/tier0/bluechi-is-online-agent-wait/test_bluechi_is_online_agent_wait.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import logging
+import os
+import threading
+import time
+from typing import Dict
+
+from bluechi_test.bluechi_is_online import BluechiIsOnline
+from bluechi_test.config import BluechiAgentConfig, BluechiControllerConfig
+from bluechi_test.machine import BluechiAgentMachine, BluechiControllerMachine
+from bluechi_test.test import BluechiTest
+from bluechi_test.util import Timeout
+
+LOGGER = logging.getLogger(__name__)
+
+NODE_FOO = "node-foo"
+IMMEDIATE_RETURN_TIMEOUT = int(os.getenv("IMMEDIATE_RETURN_TIMEOUT", 1000))
+WAIT_PARAM_VALUE = int(os.getenv("WAIT_PARAM_VALUE", 5000))
+MAX_WAIT_PARAM_VALUE = int(os.getenv("MAX_WAIT_PARAM_VALUE", 20000))
+SLEEP_DURATION = int(os.getenv("SLEEP_DURATION", 2))
+
+
+class ResultFuture:
+    def __init__(self):
+        self.result = None
+        self.output = ""
+
+
+def check_agent(
```

Comment:
Could you please simplify the method as it's used only for single use case?

```python
def check_agent(
    bluechi_is_online: BluechiIsOnline,
    wait_time: int,
    future: ResultFuture,
):
    future.result = bluechi_is_online.agent_is_online(
        wait_time,
    )
```

Reply:
Done :+1:

Reply:
Not entirely, please recheck :-;

Reply:
You're correct :) 
Done.